### PR TITLE
Added RK4 ODE solver

### PR DESCRIPTION
**Pull request by YiweiIvy**
Replaced the forward-Euler formula with a Runge-Kutta Methods of order 4. RK4, in general can generate a more accurate result compared to a forward Euler methods.

The RK4 method is implemented from scratch with pure mathematics, so that it can be easily transferred to other languages(C, C++ and so on). Note: There are some Python packages such as (Scipy.integrate.RK4) that can do the integration in one line, but may not be as transferable to other languages.

### CODE REVIEW

File: Python/src/FurutaPendulum/PendulumSimulationRK4.lf [hunk 1]
```diff
@@ -0,0 +1,180 @@
+target Python;
+/**
+ * A simple forward-Euler simulation of a Furuta pendulum,
```
Comment (lhstrh) [inline Python/src/FurutaPendulum/PendulumSimulationRK4.lf:3]:
```suggestion
 * A simple RK4 simulation of a Furuta pendulum,
```

File: Python/src/FurutaPendulum/PendulumSimulationRK4.lf [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+ * received on `d`, its value provides an instantaneous
+ * increment or decrement to `d_theta`.
+ * Notice that no output is produced when an input
+ * is received on `d` unless that input is simultaneous
+ * with the sampling period. The effect of the disturbance
+ * will not be seen on the outputs until the next sample,
+ * as would usually be the case for digital controller.
+ *
+ * @author Edward A. Lee
+ */
```
Comment (lhstrh) [inline Python/src/FurutaPendulum/PendulumSimulationRK4.lf:37]:
```suggestion
 * @author Yiwei Yu
 */
```

**Review by lhstrh — APPROVED** [review]
Looks great! There are some outdated comments that I suggested corrections for.

**Comment by lhstrh** [comment]
I think it perhaps makes more sense to use `Scipy.integrate.RK45` for the Python version, but this is a good blue print for the C version. Do you think you would be able to port it?

**Comment by YiweiIvy** [comment]
Sure, I will make changes to the python and C files accordingly.

**Review by edwardalee — APPROVED** [review]
Very cool!  One priority I have is to support methods in C and Python so that some of this duplicated code can be factored out into methods and inherited by variants of this simulator.  Methods are currently only supported in the C++ target of LF.

**Pull request closed without merging**